fix(compiler): only report parser errors on invalid expression

crisbeto opened the pull request:
Currently we reuse the same binding parser for all expressions in the template. Under the hood, the parser has a single `errors` array that it passes into all ASTs which means that if there's one binding with an error, those errors will be propagated to all other ASTs in the template.

These changes switch to having a unique `errors` array for each AST so we only report errors once.

Relates to #61792.

AndrewKushnir commented on packages/compiler/src/expression_parser/parser.ts:
```diff
@@ -113,12 +112,13 @@ export class Parser {
     ).parseChain();
 
-    return new ASTWithSource(ast, input, location, absoluteOffset, this.errors);
+    //
```
Is this change intentional?

AndrewKushnir reviewed (approved):
@crisbeto thanks for the fix!

crisbeto committed:
fix(compiler): only report parser errors on invalid expression

Currently we reuse the same binding parser for all expressions in the template. Under the hood, the parser has a single `errors` array that it passes into all ASTs which means that if there's one binding with an error, those errors will be propagated to all other ASTs in the template.

These changes switch to having a unique `errors` array for each AST so we only report errors once.

Relates to #61792.

cyber-hari commented:
LGTM!

kirjs commented:
This PR was merged into the repository by commit 5a76826d266b4ed0ef863221571e4b6b1b16182f.

The changes were merged into the following branches: main
